### PR TITLE
Add core widget base class providing module semver ranges

### LINKED ISSUE #1002: Provide other value than '*' for core widgets' [model/view]_module_version
This probably requires creating two base widgets for all core widgets... above `DOMWidget` and `Widget` respectively.

### PR DESCRIPTION
**Pull request by SylvainCorlay**
Closes #1002.

On the Python side, I use multiple inheritance to specify the version range for all core widgets without impacting the base `Widget` and `DOMWidget` classes that are used in third-party libraries.

On the JS side, I still need to make the change.

Proposed solution: use `'^' + require('../package.json').version`

@jasongrout thoughts?

### CODE REVIEW

File: ipywidgets/widgets/widget_core.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+"""Base widget class for widgets provided in Core
+
+Represents a container that can be used to group other widgets.
```
Comment (jasongrout) [inline ipywidgets/widgets/widget_core.py]: Delete this line?

**Comment by jasongrout** [comment]
> Proposed solution: use '^' + require('../package.json'),version

That sounds consistent to me.

**Comment by jasongrout** [comment]
Maybe this?
```
`^${require('package.json').version}`
```

**Comment by SylvainCorlay** [comment]
> Maybe this?
> `^${require('../package.json').version}`

This is cool :)

**Comment by SylvainCorlay** [comment]
So on the js side, I have `CoreWidgetModel` and `CoreDomWidgetModel` since `default()` will not work well with multiple inheritance.

**Comment by SylvainCorlay** [comment]
So if the semver req specified by the backend for a model is `^4.3.0`, and the installed/available js version is `4.4.3`, what happens when the backend opens a widget, is that the value for `model_module_version` stored in the model is `^4.3.0`.

When we save the widget state, this value is different from the js-side default (^4.4.3) and therefore is the one used in the serialization.

An idea would be to have a custom serializer for these attributes to restrict the semver range.

**Comment by SylvainCorlay** [comment]
@ellisonbg @jasongrout

**Comment by SylvainCorlay** [comment]
After in-person discussion with @jasongrout on this, we decided that this was still the right approach.

**Pull request merged**